comp/etw/impl/etwSession.go

```diff
@@ -166,11 +166,18 @@ func (e *etwSession) StopTracing() error {
 		globalError = errors.Join(globalError, e.DisableProvider(guid))
 	}
 	ptp := (C.PEVENT_TRACE_PROPERTIES)(unsafe.Pointer(&e.propertiesBuf[0]))
-	ret := windows.Errno(C.ControlTraceW(
-		e.hSession,
-		nil,
-		ptp,
-		C.EVENT_TRACE_CONTROL_STOP))
+	var ret windows.Errno
+	if e.wellKnown {
+		ret = windows.Errno(C.CloseTrace(e.hTraceHandle))
```

Comment:
check that `e.hTraceHandle` is valid before closing. Is it's init/error state `NULL` or `INVALID_HANDLE_VALUE` ?